de.dlr.sc.virsat.uiengine.ui/src/de/dlr/sc/virsat/uiengine/ui/editor/snippets/dialog/UiSnippetRoleManagementHandleUsersListDialog.java

```diff
@@ -0,0 +1,384 @@
+    @Override
+	protected void buttonPressed(int buttonId) {
+		if (IDialogConstants.OK_ID == buttonId) {
+			String[] updatedFeatures = featuresList.getItems();
+			notifyChanges(updatedFeatures);  // Notify with the updated features
+			okPressed();
+			close();
+		} else if (IDialogConstants.CANCEL_ID == buttonId) {
+			cancelPressed();
+		}
+	}
+    /**
+     * Creates and configures the "Update" button within the specified composite.
+     *
+     * @param buttonsComposite The composite in which the button will be created.
+     * @param valueText The Text widget containing the value to be updated.
+     */
+    private void updateButton(Composite buttonsComposite, Text valueText) {
+    	Button updateButton = new Button(buttonsComposite, SWT.PUSH);
+	    updateButton.setText("Rename");
+	    GridData updateButtonGridData = new GridData(SWT.FILL, SWT.CENTER, true, false);
+	    updateButton.setLayoutData(updateButtonGridData);
+	    
+	    // Update button listener
+        updateButton.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                String selectedValue = valueText.getText().trim();
+
+                if (selectedValue.isEmpty()) {
+                    MessageBox messageBox = new MessageBox(getShell(), SWT.ICON_INFORMATION | SWT.OK);
+                    messageBox.setMessage("Please select a feature to update.");
+                    messageBox.open();
+                    return;
+                }
+
+                int selectedIndex = featuresList.getSelectionIndex();
+
+                // Check if the selectedValue already exists in the features list
+                if (featureAlreadyExists(selectedValue)) {
+                    MessageBox messageBox = new MessageBox(getShell(), SWT.ICON_INFORMATION | SWT.OK);
+                    messageBox.setMessage("Feature already exists in the list.");
+                    messageBox.open();
+                    return;
+                }
+
+                if (selectedIndex != -1) {
+                    featuresList.setItem(selectedIndex, selectedValue);
+                    features[selectedIndex] = selectedValue;
+                    valueText.setText("");
+                }
+            }
+        });
+    }
+    /**
+     * Update the features list based on the input value
+     * 
+     *@param filter the features that contain the input value
+     */
+    private void updateFeaturesList(String value) {
+        for (String feature : features) {
+        	if (feature.contains(value) && !featureAlreadyExists(feature)) {
+                featuresList.add(feature);
+            }
+        }
+    }
+    
+    /**
+     * Checks if the given feature already exists in the features list.
+     *
+     * @param feature The feature to check for existence in the list
+     * @return True if the feature already exists, false otherwise
+     */
+    private boolean featureAlreadyExists(String feature) {
```

Comment:
The method is about users, isn't it? Why not call it userAlreayExists? Any maybe it would make sense to also hand over the list as method parameter instead of using the globale featureList? And, well, doesn't the EList implement the contains method? Then you could just do featureList.contains(feature) instead of using this method, right?

Reply:
I changed the name and called it userAlreayExists. The Elist implements the contains method but i got an error like it can not compare with the string and when i tried to cast to a string i got again an unresolved error.